lua/toggleterm/terminal.lua

```diff
@@ -121,9 +132,11 @@ local function setup_buffer_autocommands(term)
   local conf = config.get()
   api.nvim_create_augroup(AUGROUP, { clear = true })
   api.nvim_create_autocmd("TermClose", {
-    buffer = term.bufnr,
     group = AUGROUP,
-    callback = function() delete(term.id) end,
+    callback = function(args)
+      local closedTerm = find_terminal_by_bufnr(args.buf)
```

Comment:
Snake case is the convention in this project and most lua projects (that I'm aware of) i.e. should be `closed_term`